vector/src/main/java/im/vector/activity/VectorUniversalLinkActivity.java

```diff
@@ -226,6 +217,101 @@ public void onUnexpectedError(Exception e) {
+     *
+     * @param uri  the uri
+     */
+    private void renewAccountValidity(Uri uri) {
+        Log.i(LOG_TAG, "## renewAccountValidity()");
+
+        String serverUrl = uri.getScheme() + "://" + uri.getHost();
+
+        final HomeServerConnectionConfig homeServerConfig = HomeServerConnectionConfigFactoryKt.createHomeServerConnectionConfig(serverUrl, serverUrl);
+
+        String token = null;
+        Set<String> names = uri.getQueryParameterNames();
+        for (String name : names) {
+            if (ACCOUNT_VALIDITY_RENEW_TOKEN.equals(name)) {
+                String value = uri.getQueryParameter(name);
+                try {
+                    token = URLDecoder.decode(value, "UTF-8");
+                } catch (Exception e) {
+                    Log.e(LOG_TAG, "## renewAccountValidity(): Exception - parse query params Msg=" + e.getLocalizedMessage(), e);
+                }
+                break;
+            }
+        }
+
+        TchapValidityRestClient validityRestClient = new TchapValidityRestClient(homeServerConfig);
+        validityRestClient.renewAccountValidity(token, new ApiCallback<Void>() {
+
+            @Override
+            public void onSuccess(Void info) {
+                Log.i(LOG_TAG, "## renewAccountValidity() succeeded");
+                onResult(getString(R.string.tchap_renew_account_validity_success_msg), true);
+            }
+
+            @Override
+            public void onNetworkError(Exception e) {
+                onError(e.getLocalizedMessage());
+            }
+
+            @Override
+            public void onMatrixError(MatrixError e) {
+                onError(e.getLocalizedMessage());
+            }
+
+            @Override
+            public void onUnexpectedError(Exception e) {
+                onError(getString(R.string.tchap_renew_account_validity_invalid_token_msg));
+            }
+
+            private void onError(final String message) {
+                Log.e(LOG_TAG, "## renewAccountValidity() failed: " + message);
+                if (TextUtils.isEmpty(message)) {
+                    onResult(getString(R.string.tchap_error_message_default), false);
+                } else {
+                    onResult(message, false);
+                }
+            }
+
+            private void onResult(final String message, final Boolean isSuccess) {
+                new AlertDialog.Builder(VectorUniversalLinkActivity.this)
+                        .setMessage(message)
+                        .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
+                                    @Override
+                                    public void onClick(DialogInterface dialog, int which) {
+                                        if (isSuccess) {
+                                            Matrix.getInstance(VectorUniversalLinkActivity.this).onRenewAccountValidity();
+                                        }
+                                        bringAppToForeground();
```

Comment:
If you want to force user to click the ok button, it's maybe a good idea to add `.setCancellable(false)` to the dialog builder

Reply:
I don't want to force user to click on "ok", but you make me realise there was an issue in case of "back" press. So I added a dismiss listener...

Reply:
ok, cool